package.json

```diff
@@ -8,6 +8,98 @@
     ".": "dist/browser/algosdk.min.js",
     "crypto": false
   },
+  "exports": {
+    ".": {
+      "import": "./dist/esm/index.js",
+      "require": "./dist/cjs/index.js",
+      "types": "./dist/types/index.d.ts"
+    },
```

Comment:
I'm curious why you added more export paths beyond just this one. Do you think they are necessary/useful?

Our mindset has been to expose everything public through the top-level import, since that allows us to make changes to the package internal file structure without breaking anyone's imports.

I guess I'm just saying I'd rather not encourage/allow additional import paths, unless there is a compelling reason to do so.

Reply:
Completely agree, it shouldn't be encouraged unless there is a good reason. I've added them all just for completeness and we remove everything other than the critical ones. The main reasons are tree-shaking and when type inference fails. 

## Critical exports

The most important one is the `HTTPClient` models, it's very likely that we lose the inference of the original type and we need to `cast` the response type in third party libraries. 

## Nice to haves

- `client` Perfect target for tree-shaking. In projects without any interactivity they either use the `HTTPClients` or write their own custom `fetch` interfaces. Having this be stand-alone would reduce third party bundle sizes in this context. A lot of projects allow for the barrel file export (single entrypoint) along side the tree-shakeable export path

- `nacl/encoding` would be a good option for third parties to implement their own transaction factories using the custom encoding and nacl wrapper. If the third party implementation are sound, it would be much easier to adopt in the `sdk`